src/borg/testsuite/compress.py

```diff
@@ -49,168 +52,122 @@ def test_lz4_buffer_allocation(monkeypatch):
         meta1, cdata1 = c.compress({}, data)
         cdata2 = zlib.compress(data, level)
         assert cdata1 == cdata2
         meta2, data2 = c.decompress({}, cdata2)
         assert data == data2
-        # _, data2 = Compressor(**params).decompress({}, cdata2)
-        # assert data == data2
 
 
-def test_compressor():
-    params_list = [
+@pytest.mark.parametrize(
+    "c_params",
+    [
         dict(name="none"),
         dict(name="lz4"),
         dict(name="zstd", level=1),
-        dict(name="zstd", level=3),
-        # avoiding high zstd levels, memory needs unclear
+        dict(name="zstd", level=3),  # avoiding high zstd levels, memory needs unclear
         dict(name="zlib", level=0),
         dict(name="zlib", level=6),
         dict(name="zlib", level=9),
-    ]
-    if lzma:
-        params_list += [
-            dict(name="lzma", level=0),
-            dict(name="lzma", level=6),
-            # we do not test lzma on level 9 because of the huge memory needs
-        ]
-    for params in params_list:
-        c = Compressor(**params)
-        meta_c, data_compressed = c.compress({}, data)
-        assert "ctype" in meta_c
-        assert "clevel" in meta_c
-        assert meta_c["csize"] == len(data_compressed)
-        assert meta_c["size"] == len(data)
-        meta_d, data_decompressed = c.decompress(meta_c, data_compressed)
-        assert data == data_decompressed
-        assert "ctype" in meta_d
-        assert "clevel" in meta_d
-        assert meta_d["csize"] == len(data_compressed)
-        assert meta_d["size"] == len(data)
+    ] + ([
+        dict(name="lzma", level=0),
+        dict(name="lzma", level=6)
+    ] if lzma else []),  # we do not test lzma on level 9 because of the huge memory needs
+)
+def test_compressor(c_params):
+    c = Compressor(**c_params)
+    meta_c, data_compressed = c.compress({}, data)
+    assert "ctype" in meta_c
+    assert "clevel" in meta_c
+    assert meta_c["csize"] == len(data_compressed)
+    assert meta_c["size"] == len(data)
+    meta_d, data_decompressed = c.decompress(meta_c, data_compressed)
+    assert data == data_decompressed
+    assert "ctype" in meta_d
+    assert "clevel" in meta_d
+    assert meta_d["csize"] == len(data_compressed)
+    assert meta_d["size"] == len(data)
 
 
 def test_auto():
     compressor_auto_zlib = CompressionSpec("auto,zlib,9").compressor
     compressor_lz4 = CompressionSpec("lz4").compressor
     compressor_zlib = CompressionSpec("zlib,9").compressor
-    data = bytes(500)
-    meta, compressed_auto_zlib = compressor_auto_zlib.compress({}, data)
-    _, compressed_lz4 = compressor_lz4.compress({}, data)
-    _, compressed_zlib = compressor_zlib.compress({}, data)
+    data_bytes = bytes(500)
+    meta, compressed_auto_zlib = compressor_auto_zlib.compress({}, data_bytes)
+    _, compressed_lz4 = compressor_lz4.compress({}, data_bytes)
+    _, compressed_zlib = compressor_zlib.compress({}, data_bytes)
```

Comment:
i found `data` and `cdata` an nice and short way to refer to uncompressed and compressed data.

don't see much advantage in using `data_bytes`.

Reply:
I agree with your logic, and only changed to prevent confusion with `data` variable from outer scope. I can revert this change :)

Reply:
yes, please!

if there is a problem with that shadowing, just rename the global to e.g. DATA.